src/OMPLConversions.cpp

```diff
@@ -135,16 +137,35 @@ ompl::base::StateSpacePtr CreateStateSpace(OpenRAVE::RobotBasePtr const robot,
     ompl::base::RealVectorBounds bounds(num_dof);
     for (size_t i = 0; i < num_dof; ++i) {
         BOOST_ASSERT(lowerLimits[i] <= upperLimits[i]);
+        if (is_continuous[i])
+        {
+            double diam = upperLimits[i] - lowerLimits[i];
+            if (fabs(diam - 2.0*M_PI) > std::numeric_limits<double>::epsilon() * 2.0)
+            {
+                RAVELOG_WARN("Robot DOF [%lu] is circular, but has limits [%f,%f]!\n",
+                    i, lowerLimits[i], upperLimits[i]);
+                RAVELOG_WARN("Ignoring limits and using [-PI,PI] instead ...\n");
+                lowerLimits[i] = -M_PI;
+                upperLimits[i] = M_PI;
+            }
+        }
         bounds.setLow(i, lowerLimits[i]);
         bounds.setHigh(i, upperLimits[i]);
     }
     
     // construct state space
     ompl::base::StateSpacePtr state_space;
     if (any_continuous) {
+#if 0
         state_space.reset(new ContinuousJointsStateSpace(is_continuous));
         RAVELOG_DEBUG("Setting joint limits.\n");
         state_space->as<ContinuousJointsStateSpace>()->setBounds(bounds);
+#else
+        state_space.reset(new SemiTorusStateSpace(num_dof));
+        state_space->as<SemiTorusStateSpace>()->setIsWrapping(is_continuous);
+        RAVELOG_DEBUG("Setting joint limits.\n");
+        state_space->as<SemiTorusStateSpace>()->setBounds(bounds);
+#endif
```

Comment:
Please remove the dead code that creates `ContinuousJointsStateSpace`, since it has been replaced by `SemiTorusStateSpace`. If this is the only reference to that code, then we should also delete the implementation itself.